src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs

```diff
@@ -54,7 +54,7 @@ public NuGetFramework TargetFramework
         public string Config => AssemblyFullPath + ".config";
         public string DepsJson => Path.Combine(TargetDirectory, Path.GetFileNameWithoutExtension(AssemblyFullPath) + ".deps.json");
         public string RuntimeConfigJson => Path.Combine(TargetDirectory, Path.GetFileNameWithoutExtension(AssemblyFullPath) + ".runtimeconfig.json");
-
+        public string PackageLockFile => Path.Combine(Path.GetDirectoryName(ProjectFullPath), "obj", "project.assets.json");
```

Comment:
Good point. Dug a little deeper. Looks like it could be in a different folder.

cref https://github.com/NuGet/NuGet.Client/blob/971f47f885db1afcb7845411df38b1212e4151b0/src/NuGet.Core/NuGet.ProjectModel/RestoreOutputType.cs#L15-L18
https://github.com/NuGet/NuGet.Client/blob/1c1d537b379936a76dcc2b010fdfdecb54657070/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L315